HashRegistrarSimplified.sol

```diff
@@ -286,30 +318,23 @@ contract Registrar {
         if (address(bid) == 0 ) throw;
         sealedBids[seal] = Deed(0);
         bid.setOwner(_owner);
-        entry h = entries[_hash];
+        entry h = _entries[_hash];
         uint actualValue = min(_value, bid.balance);
-        bid.setBalance(actualValue);            
-
-        
-        if (h.status == Mode.Forbidden) {
-            /* 
-            If during the auction process someone reveals that the name being bidded
-            on is invalid (shorter than 7 chars) then all bids will be fully refunded
-            */
-            bid.closeDeed(1000);
-            BidRevealed(_hash, _owner, actualValue, 4);
-        } else if (bid.creationDate() > h.registrationDate - revealPeriod
-            || now > h.registrationDate 
-            || actualValue < minPrice) {
-            /* 
-            A penalty (99%) is applied for submitting unrevealed bids, which could otherwise
-            be used as a threat of revealing a bid higher than the second-highest 
-            bid, to extort the winner into paying them.
-            */
-            bid.closeDeed(10);
+        bid.setBalance(actualValue);
+
+        var auctionState = state(_hash);
+        if(auctionState == Mode.Owned) {
```

Comment:
If we allow auctions to be invalidated at any point, then we could add a `if invalid` that would refund 100% of bids